test/contracts/common/erc721-transfer.ts

```diff
@@ -0,0 +1,84 @@
+import {BigNumber, Event} from 'ethers'
+import {TransferEvent} from '../../../typechain-types/IERC721'
+import {expect} from 'chai'
+import {Result} from '@ethersproject/abi'
+
+export function deepEqualsERC721TokenTransfer(
+    actual: ActualERC721Transfer,
+    expected: ExpectedERC721Transfer
+): boolean {
+    return (
+        actual.to === expected.to &&
+        actual.from === expected.from &&
+        actual.tokenId.toBigInt() === expected.tokenId
+    )
+}
+
+/**
+ * Shape check and conversion for a TransferEvents.
+ */
+export function erc721TransferEvents(_events: Event[]): {
+    from: string
+    to: string
+    tokenId: BigNumber
+}[] {
+    const converted: {
```

Comment:
suggestion: code reuse, pull the anonymous object into a type scoped to this file.